### PR TITLE
Add bbe for group by/collect

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose
$subject

### CODE REVIEW

File: examples/aggregation/aggregation.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Aggregation
+
+The `group by` clause in the query expression can be used to group the elements in a collection. Grouping happens based on the grouping keys provided in `group by` clause. Non grouping keys becomes aggregated variables that can be used as a list or an argument to a rest parameter of a langlib function.
+
+The `collect` clause is used to collect the collection into one group. All the variables becoes aggregated variables and those variables can be used as a list or an argument to a rest parameter of a langlib function same as in `group by`.
```
Comment (gimantha) [inline examples/aggregation/aggregation.md:5]:
```suggestion
The `collect` clause is used to collect the collection into one group. All the variables become aggregated variables and those variables can be used as a list or an argument to a rest parameter of a langlib function same as in `group by`.
```

File: examples/aggregation/aggregation.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Aggregation
```
Comment (gimantha) [inline examples/aggregation/aggregation.md:1]: I think it is better to give a small description on what is meant by grouping/non grouping keys and sequence variables

**Review by gimantha — APPROVED** [review]
LGTM

File: examples/aggregation/aggregation.bal [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        {orderId: 1, itemName: "B", price: 21.5, quantity: 3}
+    ];
+
+    var items = from var {orderId, itemName} in orders
+        // The `group by` clause create groups for each `orderId`.
+        // The `itemName` is a non-grouping key and it becomes a sequence variable.
+        group by orderId
+        select [itemName];
+    
+    // List of items per `orderId`
```
Comment (praneesha) [inline examples/aggregation/aggregation.bal:17]:
```suggestion
    // List of items per `orderId`.
```

File: examples/aggregation/aggregation.bal [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    io:println(items);
+
+    var quantities = from var {itemName, quantity} in orders
+        // The `group by` clause create groups for each `itemName`.
```
Comment (praneesha) [inline examples/aggregation/aggregation.bal:21]:
```suggestion
        // The `group by` clause creates the groups for each `itemName`.
```

File: examples/aggregation/aggregation.bal [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        group by itemName
+        select {itemName, quantity: sum(quantity)};
+
+    // List of quantity per item
+    io:println(quantities);
+
+    var income = from var {price, quantity} in orders
+        let var totPrice = price*quantity
+        // The `collect` clause creates a single group and all variables become
+        // non-grouping keys
```
Comment (praneesha) [inline examples/aggregation/aggregation.bal:32]:
```suggestion
        // non-grouping keys.
```

File: examples/aggregation/aggregation.bal [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        collect sum(totPrice);
+    
+    // Total Income from orders
```
Comment (praneesha) [inline examples/aggregation/aggregation.bal:35]:
```suggestion
    // Calculate the total income from all the orders.
```

File: examples/aggregation/aggregation.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+The `group by` clause in the query expression can group the elements in a collection. Grouping happens based on the grouping keys provided in `group by` clause. For each group, grouping keys are unique. All other variables other than grouping keys are called non-grouping keys. For each group, non-grouping keys become sequence variables. Those variables can be used as a list or an argument to a rest parameter of a langlib function.
```
Comment (praneesha) [inline examples/aggregation/aggregation.md:3]:
```suggestion
The `group by` clause in a query expression can group the elements in a collection. Grouping happens based on the grouping keys provided in the `group by` clause. Grouping keys will be unique for each group. All variables other than the grouping keys are called non-grouping keys. For each group, non-grouping keys become sequence variables. Those variables can be used as a list or an argument to a rest parameter of a langlib function.
```

File: examples/aggregation/aggregation.metatags [hunk 8]
```diff
@@ -0,0 +1,2 @@
+description: This BBE demonstrates how to group a collection, how to handle non grouping keys.
```
Comment (praneesha) [inline examples/aggregation/aggregation.metatags:1]:
```suggestion
description: This BBE demonstrates how to group a collection and how to handle non-grouping keys.
```

**Review by praneesha — APPROVED** [review]

**Review by gimantha — APPROVED** [review]

**Pull request merged**